src/lib/sw_engine/tvgSwShape.cpp

```diff
@@ -254,7 +254,7 @@ static void _dashLineTo(SwDashStroke& dash, const Point* to, const Matrix* trans
             Line left, right;
             _lineSplitAt(cur, dash.curLen, left, right);;
             dash.curIdx = (dash.curIdx + 1) % dash.cnt;
-            if (!dash.curOpGap) {
+            if (!dash.curOpGap && dash.curLen > FLT_EPSILON) {
```

Comment:
For a pattern like (0,0,3,4) there is no need to add to the outline points for a zero length curve.